app/javascript/app/components/map/map-component.jsx

```diff
@@ -0,0 +1,298 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { format } from 'd3-format';
+import cx from 'classnames';
+import {
+  ComposableMap,
+  ZoomableGroup,
+  Geographies,
+  Geography
+} from 'react-simple-maps';
+import { Motion, spring } from 'react-motion';
+import ReactTooltip from 'react-tooltip';
+import { Icon, Button } from 'cw-components';
+import { TabletLandscape } from 'components/responsive';
+import mapZoomIn from 'assets/icons/map-zoom-in';
+import mapZoomOut from 'assets/icons/map-zoom-out';
+import startCase from 'lodash/startCase';
+import toLower from 'lodash/toLower';
+
+import styles from './map-styles.scss';
+
+class MapComponent extends Component {
+  componentDidMount() {
+    setTimeout(
+      () => {
+        ReactTooltip.rebuild();
+      },
+      100
+    );
+  }
+
+  render() {
+    const {
+      forceUpdate,
+      paths,
+      cache,
+      style,
+      zoomEnable,
+      dragEnable,
+      tooltipId,
+      handleZoomIn,
+      handleZoomOut,
+      countryNameTooltip,
+      onCountryClick,
+      onCountryMove,
+      onCountryEnter,
+      onCountryLeave,
+      onCountryFocus,
+      onCountryBlur,
+      defaultStyle,
+      controlPosition,
+      zoom,
+      center,
+      customCenter,
+      className,
+      theme
+    } = this.props;
+
+    const getTooltip = name => {
+      const path = paths.find(p => p.properties.name === name);
```

Comment:
I think we can use destructuring here like for whole return method below:
`const { tooltipValue, tooltipUnit, sector, selectedYear } = path.properties;`